data/glsk/glsk-document-cse/src/test/java/com/farao_community/farao/data/glsk/cse/CseGlskDocumentImporterTest.java

```diff
@@ -0,0 +1,166 @@
+
+        assertNotNull(propGlskScalable);
+        assertEquals(2000., network.getGenerator("FFR1AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(2000., network.getGenerator("FFR2AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(3000., network.getGenerator("FFR3AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(1000., network.getLoad("FFR1AA1 _load").getP0(), EPSILON);
+        assertEquals(3500., network.getLoad("FFR2AA1 _load").getP0(), EPSILON);
+
+        propGlskScalable.scale(network, 1000.);
+        assertEquals(2200., network.getGenerator("FFR1AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(2200., network.getGenerator("FFR2AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(3300., network.getGenerator("FFR3AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(933.3334, network.getLoad("FFR1AA1 _load").getP0(), EPSILON);
+        assertEquals(3266.6667, network.getLoad("FFR2AA1 _load").getP0(), EPSILON);
+    }
+
+    @Test
+    public void checkCseGlskDocumentImporterCorrectlyImportMeritOrderGskBlocks() {
+        CseGlskDocument cseGlskDocument = CseGlskDocument.importGlsk(getClass().getResourceAsStream("/testGlsk.xml"));
+        List<AbstractGlskPoint> list = cseGlskDocument.getGlskPoints("FR_MERITORDER");
+        assertFalse(list.isEmpty());
+        assertEquals(1, list.size());
+        assertEquals(4, list.get(0).getGlskShiftKeys().size());
+        assertEquals(1, list.get(0).getGlskShiftKeys().get(0).getRegisteredResourceArrayList().size());
+        assertEquals(1, list.get(0).getGlskShiftKeys().get(1).getRegisteredResourceArrayList().size());
+        assertEquals(1, list.get(0).getGlskShiftKeys().get(2).getRegisteredResourceArrayList().size());
+        assertEquals(1, list.get(0).getGlskShiftKeys().get(3).getRegisteredResourceArrayList().size());
+    }
+
+    @Test
+    public void checkCseGlskDocumentImporterCorrectlyConvertMeritOrderGskBlocks() {
+        Network network = Importers.loadNetwork("testCase.xiidm", getClass().getResourceAsStream("/testCase.xiidm"));
+        GlskDocument glskDocument = GlskDocumentImporters.importGlsk(getClass().getResourceAsStream("/testGlsk.xml"));
+        Scalable propGskScalable = glskDocument.getZonalScalable(network).getData("FR_MERITORDER");
+
+        assertNotNull(propGskScalable);
+        assertEquals(2000., network.getGenerator("FFR1AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(2000., network.getGenerator("FFR2AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(3000., network.getGenerator("FFR3AA1 _generator").getTargetP(), EPSILON);
+
+        propGskScalable.scale(network, 5000.);
+        assertEquals(5000., network.getGenerator("FFR1AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(4000., network.getGenerator("FFR2AA1 _generator").getTargetP(), EPSILON);
+        assertEquals(3000., network.getGenerator("FFR3AA1 _generator").getTargetP(), EPSILON);
+    }
```

Comment:
Maybe add another test that goes to 9000 to see if the second FR1 is correctly taken into account.

Another question by curiosity here, what if scaling is higher than what is available in merit order ? PowsyBl error, logs ?

Reply:
If scale asked above max available, scale() will return a value that is different from the quantity asked (not tested here)
https://javadoc.io/static/com.powsybl/powsybl-core/3.0.0/com/powsybl/action/util/Scalable.html#scale-com.powsybl.iidm.network.Network-double-com.powsybl.action.util.Scalable.ScalingConvention-